ingestion/src/metadata/ingestion/ometa/auth_provider.py

```diff
@@ -109,3 +112,24 @@ def auth_token(self) -> str:
         }
         token = jwt.encode(claims, my_jwk.to_dict(), JWT.HASH_ALGORITHM)
         return token
+
+
+class Auth0AuthenticationProvider(AuthenticationProvider):
+    def __init__(self, config: MetadataServerConfig):
+        self.config = config
+
+    @classmethod
+    def create(cls, config: MetadataServerConfig):
+        return cls(config)
+
+    def auth_token(self) -> str:
+        conn = http.client.HTTPSConnection(self.config.domain)
+        payload = "grant_type=client_credentials" + "&client_id=" + self.config.client_id + "&client_secret=" \
```

Comment:
can we use python string formating. Please look at other code we use like get_sql_connection_url